eclipse/src/saros/ui/wizards/AddProjectToSessionWizard.java

```diff
@@ -666,4 +680,79 @@ private void displaySaveDirtyEditorsDialog(final List<IEditorPart> dirtyEditors)
       for (IEditorPart editor : dirtyEditors) editor.doSave(new NullProgressMonitor());
     }
   }
+
+  private void storeCurrentProjectNameMapping(final JID jid) {
+    Map<String, String> currentProjectNameMapping = new HashMap<>();
+
+    for (final ProjectNegotiationData data : negotiation.getProjectNegotiationData()) {
+      final String projectID = data.getProjectID();
+      currentProjectNameMapping.put(
+          data.getProjectName(), namePage.getTargetProjectName(projectID));
+    }
+
+    setNewProjectNameMapping(jid, currentProjectNameMapping);
+  }
+
+  private Map<String, String> getLastProjectNameMapping(final JID jid) {
+    Map<String, String> result = new HashMap<>();
+
+    final String mapping = preferenceStore.getString(projectMappingStorePrefix + jid.getBase());
+
+    if (mapping == null || mapping.isEmpty()) return result;
+
```

Comment:
the following code is very complex. Why not:
```java
// split only at unescaped colons
String[] splitted = res.split("(?<!\\\\):");
for (int i = 0; i < splitted.length; i += 2 )
{
  String key = splitted[i].replace("\\:", ":");
  String value = splitted[i + 1].replace("\\:", ":");
  result.put(key, value);  
}
```

Reply:
Thanks ! It the split really Java Syntax ? Isn't it (?!<[^\\\\]):

Reply:
Yes this is Java Syntax. I tested the code with a simple example (executed with jdk8):
```java
import java.util.List;
import java.util.Arrays;
import java.util.Map;
import java.util.HashMap;

public class Test {
  public static void main(String[] args) {
    String res = "\\:remote\\:1:\\:loca\\:l1:remote2:local2\\:";
    Map<String, String> result = new HashMap<>();

    String[] splitted = res.split("(?<!\\\\):");
    for (int i = 0; i < splitted.length; i+=2 ) {
      String key = splitted[i].replace("\\:", ":");
      String value = splitted[i + 1].replace("\\:", ":");
      result.put(key, value);  
    }

    System.out.println(result.get(":remote:1"));
    System.out.println(result.get("remote2"));
  }
}

```